frame/vesting/src/lib.rs

```diff
@@ -300,21 +358,32 @@ impl<T: Config> VestedTransfer for Pallet<T> {
 	type Moment = MomentOf<T>;
 	type Balance = BalanceOf<T>;
 	type MinVestedTransfer = T::MinVestedTransfer;
+	type VestingScheduleId = T::VestingScheduleId;
+	type VestingScheduleCount = VestingScheduleCount<T>;
 
 	#[transactional]
 	fn vested_transfer(
 		asset: Self::AssetId,
 		from: &Self::AccountId,
 		to: &Self::AccountId,
-		schedule: VestingSchedule<Self::BlockNumber, Self::Moment, Self::Balance>,
+		schedule_input: VestingScheduleInput<Self::BlockNumber, Self::Moment, Self::Balance>,
 	) -> frame_support::dispatch::DispatchResult {
 		ensure!(from != to, Error::<T>::TryingToSelfVest);
 
+		let vesting_schedule_id = Self::VestingScheduleCount::increment()?; // VestingScheduleCount::<T>::increment()?;
+		let schedule: VestingScheduleOf<T> = VestingSchedule {
+			vesting_schedule_id,
+			window: schedule_input.window,
+			period_count: schedule_input.period_count,
+			per_period: schedule_input.per_period,
+			already_claimed: Zero::zero(),
+		};
```

Comment:
perhaps an associated function on `VestingSchedule`, `fn from_input(id, input)`? would make this cleaner and be useful in tests as well